### DISCUSSION TITLE
Error installing llama-cpp-pythin on Windows | Pls help

### DISCUSSION THREAD

**Discussion by arjuns2020** [abetlen/llama-cpp-python#1001] [Q&A]
pip install llama-cpp-python --no-cache-dir
Collecting llama-cpp-python
  Downloading llama_cpp_python-0.2.22.tar.gz (8.7 MB)
     ---------------------------------------- 8.7/8.7 MB 4.1 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: typing-extensions>=4.5.0 in c:\users\arjun\anaconda3\envs\privategpt\lib\site-packages (from llama-cpp-python) (4.8.0)
Requirement already satisfied: numpy>=1.20.0 in c:\users\arjun\anaconda3\envs\privategpt\lib\site-packages (from llama-cpp-python) (1.26.0)
Requirement already satisfied: diskcache>=5.6.1 in c:\users\arjun\anaconda3\envs\privategpt\lib\site-packages (from llama-cpp-python) (5.6.3)
Building wheels for collected packages: llama-cpp-python
  Building wheel for llama-cpp-python (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for llama-cpp-python (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      *** scikit-build-core 0.7.0 using CMake 3.27.9 (wheel)
      *** Configuring CMake...
      2023-12-11 22:27:04,430 - scikit_build_core - WARNING - Can't find a Python library, got libdir=None, ldlibrary=None, multiarch=None, masd=None
      loading initial cache file C:\Users\arjun\AppData\Local\Temp\tmpmrvmqll4\build\CMakeInit.txt
      -- Building for: NMake Makefiles
      CMake Error at CMakeLists.txt:3 (project):
        Running

         'nmake' '-?'

        failed with:

         The system cannot find the file specified

      CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
      CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
      -- Configuring incomplete, errors occurred!

      *** CMake configuration failed
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for llama-cpp-python
Failed to build llama-cpp-python
ERROR: Could not build wheels for llama-cpp-python, which is required to install pyproject.toml-based projects

**Comment by tomshaffner** [comment]
Do you have CMake installed and set? It looks from your error above like you don't; would need to do that first.

**Reply by arjuns2020** [reply]
I Saw this article 

https://stackoverflow.com/questions/50345127/cmake-mingw-unknown-compilers-gcc-exe-broken

I specified the compiler cmake -DCMAKE_C_COMPILER="C:\path\to\gcc.exe" -DCMAKE_CXX_COMPILER="C:\path\to\g++.exe"

tried this 
pip install llama-cpp-python --global-option="--cpp_ext" --global-option="--cmake_args=-DCMAKE_C_COMPILER=C:/path/to/compiler/gcc.exe -DCMAKE_CXX_COMPILER=C:/path/to/compiler/g++.exe"


I installed & configured gcc , c++, nmake & cmake

I have set path for all the compiler ,nmake,wingw,conda etc..


still I get this error 

  × Building wheel for llama-cpp-python (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      *** scikit-build-core 0.7.0 using CMake 3.28.0 (wheel)
      *** Configuring CMake...
      2023-12-13 22:50:01,217 - scikit_build_core - WARNING - Can't find a Python library, got libdir=None, ldlibrary=None, multiarch=None, masd=None
      loading initial cache file C:\Users\arjun\AppData\Local\Temp\tmpwnt18__5\build\CMakeInit.txt
      -- Building for: NMake Makefiles
      CMake Error at CMakeLists.txt:3 (project):
        Running

         'nmake' '-?'

        failed with:

         no such file or directory


      CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
      CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
      -- Configuring incomplete, errors occurred!

      *** CMake configuration failed
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for llama-cpp-python
Failed to build llama-cpp-python
ERROR: Could not build wheels for llama-cpp-python, which is required to install pyproject.toml-based projects
